File: src/main/java/io/github/gravetii/scene/game/GameEndResultComponent.java
package io.github.gravetii.scene.game;

import io.github.gravetii.controller.GameEndResultController;
import io.github.gravetii.model.GameStats;
import io.github.gravetii.scene.FxComponent;

import javafx.scene.layout.VBox;

public class GameEndResultComponent extends FxComponent<GameEndResultController, VBox> {
  private final GameGridComponent ref;

  public GameEndResultComponent(GameGridComponent ref) throws Exception {
    super("gameEndResult.fxml");
    this.ref = ref;
    this.create();
    this.updateStats();
  }

  @Override
  protected GameEndResultController createController() {
    return new GameEndResultController(this.ref.getController());
  }

  @Override
  protected VBox createNode() throws Exception {
    return this.loadNode();
  }

  private void updateStats() {
    GameStats stats = this.ref.computeStats();
    this.getController().updateText(stats.toString());
  }
}

File: src/main/kotlin/io/github/gravetii/scene/game/GameEndResultComponent.kt
package io.github.gravetii.scene.game

import io.github.gravetii.controller.GameEndResultController
import io.github.gravetii.scene.FxComponent
import javafx.scene.layout.VBox

class GameEndResultComponent(private val ref: GameGridComponent) :
    FxComponent<GameEndResultController, VBox>("gameEndResult.fxml") {

    override val controller: GameEndResultController = GameEndResultController(ref.controller)

    override val node: VBox = loadNode()

    init {
        val stats = ref.computeStats()
        controller.updateText(stats.toString())
    }

}